Realm/RLMObjectSchema.mm

```diff
@@ -132,6 +132,21 @@ + (instancetype)schemaForObjectClass:(Class)objectClass {
         }
     }
 
+    for (RLMProperty *prop in schema.properties) {
+        RLMPropertyType type = prop.type;
+        if (prop.optional && !RLMPropertyTypeIsNullable(type)) {
+#ifdef REALM_ENABLE_NULL
+            NSString *error = [NSString stringWithFormat:@"Only 'string', 'binary', and 'object' properties can be made optional, and property '%@' is of type '%@'.", prop.name, RLMTypeToString(type)];
+#else
+            NSString *error = [NSString stringWithFormat:@"Only 'object' properties can be made optional, and property '%@' is of type '%@'.", prop.name, RLMTypeToString(type)];
+#endif
+            if (prop.type == RLMPropertyTypeAny && isSwift) {
+                error = [error stringByAppendingString:@"\nIf this is a 'String?' property, it must be declared as 'NSString?' instead."];
```

Comment:
I'd presume so? I'll double check once this is merged and the `swift-2.0` branch is rebased